### PR TITLE
[Build] Show useful error message and directions for users lacking requirements to run Stride.Core.Tasks

### PR DESCRIPTION
**Pull request by Eideren**
# PR Details
Here's how it looks now when building without the right SDK installed:
image

## Related Issue
None logged in github as far as I could find

## Types of changes
- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by tebjan** [comment]
How about also outputting the download link?

**Comment by Eideren** [comment]
Done, users will have to do with a generic link though

**Review by manio143 — APPROVED** [review]

File: sources/core/Stride.Core.Tasks/Program.cs [hunk 1]
```diff
@@ -16,7 +16,20 @@ static class Program
...
+                MSBuildLocator.RegisterDefaults();
+            }
+            catch (InvalidOperationException e) when (e.Message.StartsWith("No instances of MSBuild could be detected."))
+            {
+                // When tasks running through build tools throw, it logs an obtuse 'The command [...]/Stride.Core.Tasks.exe [...] exited with code - x'
+                // message requiring the user to dig in the output to figure out what happened.
+                // The following ensures that a clear message is logged before the stuff above is shown.
+                // Do note that the 'error' word in the message is essential for it to be logged,
+                // direct any message about this peculiar 'feature' over to Microsoft, thanks !
+                Console.Error.WriteLine($@"error {typeof(Program).Namespace}: No supported instance of MSBuild could be detected, make sure you have .Net SDK {Environment.Version.Major} {Environment.Version.Minor} installed");
```
[the remark this answers is not in the document]
Comment (Eideren) [reply sources/core/Stride.Core.Tasks/Program.cs]: Yep, that's the point, the locator filters based on the environment version:
https://github.com/microsoft/MSBuildLocator/blob/master/src/MSBuildLocator/DotNetSdkLocationHelper.cs#L58-L62

**Comment by Eideren** [comment]
Rebased to trigger the teamcity builds again, base commit was breaking them I'm pretty sure

**Pull request merged**